### PR TITLE
Added Convention To Style Guide To Always Declare Functions That Take No Arguments as (void) Not ()

### PR DESCRIPTION
**Pull request by garethellis0**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — CHANGES_REQUESTED** [review]

**Review by akhilveeraghanta — COMMENTED** [review]
This doesn't seem to be currently enforced, is this more of a "moving forward" rule?

**Comment by garethellis0** [comment]
> This doesn't seem to be currently enforced, is this more of a "moving forward" rule?

Yep, we should fixes cases of this as we see them.

File: docs/code-style-guide.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ The vast majority of the things noted in this document will apply to `C` code as
   typedef struct MyDog MyDog_t;
   ```
 
+* Functions that take no arguments must be declared as `foo(void)` **not** `foo()`, as the second option allows `foo` to take anything as it's arguments ([reference](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/286494))
```
[the remark this answers is not in the document]
Comment (garethellis0) [reply docs/code-style-guide.md:61]: I'd prefer to only have one reference, do you strictly prefer the reference you linked over the one that's in there right now?

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :tada:

File: docs/code-style-guide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanlew) [reply docs/code-style-guide.md:61]: It is better but I'm not that concerned

**Review by jonathanlew — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]
LGTM

**Pull request merged**